#! /bin/sh /usr/share/dpatch/dpatch-run
## myplugin_strcpy.dpatch by Sebastian Harl <sh@tokkee.org>
##
## DP: examples/myplugin.c: Use sstrncpy instead of strcpy.

@DPATCH@

diff a/contrib/examples/myplugin.c b/contrib/examples/myplugin.c
--- a/contrib/examples/myplugin.c
+++ b/contrib/examples/myplugin.c
@@ -98,8 +98,8 @@ static int my_read (void)
 	vl.values     = values;
 	vl.values_len = 1;
 	vl.time       = time (NULL);
-	strcpy (vl.host, hostname_g);
-	strcpy (vl.plugin, "myplugin");
+	sstrncpy (vl.host, hostname_g, sizeof (vl.host));
+	sstrncpy (vl.plugin, "myplugin", sizeof (vl.plugin));
 	/* optionally set vl.plugin_instance and vl.type_instance to reasonable
 	 * values (default: "") */
 
